lib/Conversion/HWToBTOR2/HWToBTOR2.cpp

```diff
@@ -770,13 +773,63 @@ struct ConvertHWToBTOR2Pass
     // Genrate the bad btor2 intruction
     genBad(op);
   }
+
   // Assumptions are converted to a btor2 constraint instruction
   void visitSV(sv::AssumeOp op) {
     // Extract the expression that we want our constraint to be about
     Value expr = op.getExpression();
     genConstraint(expr);
   }
 
+  // Assertions are sometimes encoded using a combination of fatal and error, we
+  // thus need to find the parenting condition and generate a bad statement
+  // using that.
+  void visitEncodedAssert(Operation *op) {
+    genSort("bitvec", 1);
+
+    // Fatal simply fails, so we need to find in which case we are failing
+    // For some reason Chisel assertions are sometimes encoded using a fatal and
+    // an error.
+    if (auto ifop = dyn_cast<sv::IfOp>(((Operation *)op)->getParentOp())) {
```

Comment:
Strictly speaking I guess we need to check that its parent always statement has a same clock seen in the walk?

Reply:
There is an assumption of a single clock for now, that's specified in the clock lowering.